docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1145 @@
+  #
+  # {plugintype} {
+  #   {pluginname} { message => "My message..." }
+  # }
+  config_name "example"
+
+  # Replace the message with this value.
+  config :message, :validate => :string, :default => "Hello World!"
+
+
+  public
+  def register
+    # Add instance variables
+  end # def register
+
+  public
+  def filter(event)
+
+    if @message
+      # Replace the event message with our message as configured in the
+      # config file.
+      event["message"] = @message
+    end
+
+    # filter_matched should go in the last line of our successful code
+    filter_matched(event)
+  end # def {plugintype}
+
+end # class LogStash::{pluginclass}::{pluginnamecap}
+----------------------------------
+endif::filter_method[]
+
+// /////////////////////////////////////////////////////////////////////////////
+// Output (conditionally recognized by the presence of the receive_method
+// attribute)
+// /////////////////////////////////////////////////////////////////////////////
+ifdef::receive_method[]
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+# encoding: utf-8
+require "logstash/{plugintype}s/base"
+require "logstash/namespace"
+
+# Add any asciidoc formatted documentation here
+# An example output that does nothing.
+class LogStash::{pluginclass}::{pluginnamecap} < LogStash::{pluginclass}::Base
+  config_name "example"
+
+  public
+  def register
+  end # def register
+
+  public
+  def receive(event)
+  end # def receive
+
+end # class LogStash::{pluginclass}::{pluginnamecap}
+----------------------------------
+endif::receive_method[]
+
+== Coding {plugintype} plugins
+
+Now let's take a line-by-line look at the example plugin.
+
+=== `encoding`
+
+It seems like a small thing, but remember to specify the encoding at the
+beginning of your plugin code:
+
+[source,sh]
+----------------------------------
+# encoding: utf-8
+----------------------------------
+
+Logstash depends on things being in UTF-8, so we put this here to tell the Ruby
+interpreter that we’re going to be using the UTF-8 encoding.
+
+=== `require` Statements
+
+Logstash {plugintype} plugins require parent classes defined in
+`logstash/pass:attributes[{plugintype}]s/base` and `logstash/namespace`:
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+require "logstash/{plugintype}s/base"
+require "logstash/namespace"
+----------------------------------
+
+Of course, the plugin you build may depend on other code, or even gems. Just put
+them here along with these Logstash dependencies.
+
+=== Plugin Body
+
+Let's go through the various elements of the plugin itself.
+
+==== Inline Documentation
+Logstash provides infrastructure to automatically generate documentation for
+plugins. We use the asciidoc format to write documentation so _any_ comments in
+the source code will be first converted into asciidoc and then into html.
+
+All plugin documentation is then rendered and placed in
```

Comment:
this link doesn't seem to work... perhaps try:
http://www.elasticsearch.org/guide/en/logstash/current/index.html